src/components/network-map-tab.js

```diff
@@ -0,0 +1,351 @@
+import NetworkMap from './network/network-map';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import {
+    fetchLinePositions,
+    fetchSubstationPositions,
+} from '../utils/rest-api';
+import GeoData from './network/geo-data';
+import { equipments } from './network/network-equipments';
+import withLineMenu from './menu/line-menu';
+import BaseEquipmentMenu from './menu/base-equipment-menu';
+import withEquipmentMenu from './menu/equipment-menu';
+import VoltageLevelChoice from './voltage-level-choice';
+import LoaderWithOverlay from './util/loader-with-overlay';
+import NominalVoltageFilter from './network/nominal-voltage-filter';
+import { makeStyles } from '@material-ui/core/styles';
+import OverloadedLinesView from './network/overloaded-lines-view';
+import { RunButtonContainer } from './run-button-container';
+import { getLoadFlowRunningStatus } from './util/running-status';
+import { useSelector } from 'react-redux';
+import { PARAM_DISPLAY_OVERLOAD_TABLE } from '../utils/config-params';
+import { getLineLoadingZone, LineLoadingZone } from './network/line-layer';
+
+const INITIAL_POSITION = [0, 0];
+
+const useStyles = makeStyles((theme) => ({
+    divNominalVoltageFilter: {
+        position: 'absolute',
+        right: 10,
+        bottom: 30,
+    },
+    divRunButton: {
+        position: 'absolute',
+        right: 100,
+        bottom: 30,
+        marginLeft: 8,
+        marginRight: 8,
+        marginTop: 8,
+    },
+    divOverloadedLineView: {
+        right: 45,
+        top: 10,
+        minWidth: '500px',
+        position: 'absolute',
+        height: '70%',
+        opacity: '1',
+        flex: 1,
+        pointerEvents: 'none',
+    },
+}));
+
+export const NetworkMapTab = ({
+    /* redux can be use as redux*/
+    studyUuid,
+    network,
+    selectedNodeUuid,
+    /* results*/
+    securityAnalysisStatus,
+    runnable,
+    loadFlowInfos,
+    /* visual*/
+    visible,
+    useName,
+    lineFullPath,
+    lineParallelPath,
+    lineFlowMode,
+    lineFlowColorMode,
+    lineFlowAlertThreshold,
+    updatedLines,
+    /* callbacks */
+    openVoltageLevel,
+    setIsComputationRunning,
+    filteredNominalVoltages,
+    centerOnSubstation,
+    showInSpreadsheet,
+    setErrorMessage,
+}) => {
+    const [waitingLoadGeoData, setWaitingLoadGeoData] = useState(true);
+
+    const displayOverloadTable = useSelector(
+        (state) => state[PARAM_DISPLAY_OVERLOAD_TABLE]
+    );
+
+    const [geoData, setGeoData] = useState();
+
+    const mapRef = useRef();
+
+    const [equipmentMenu, setEquipmentMenu] = useState({
+        position: [-1, -1],
+        equipment: null,
+        equipmentType: null,
+        display: null,
+    });
+
+    const [
+        choiceVoltageLevelsSubstationId,
+        setChoiceVoltageLevelsSubstationId,
+    ] = useState(null);
+
+    const [position, setPosition] = useState([-1, -1]);
+
+    useEffect(() => {
```

Comment:
Does this `useEffect` is justified ? seems strange

Reply:
not mine :P but yes it is used to center on substation but not lock the map on it.